product/api.go

```diff
@@ -148,3 +149,17 @@ type FileInfoStoreService interface {
 type CloudService interface {
 	GetCloudLimits() (*model.ProductLimits, error)
 }
+
+// KVStoreService is the API for accessing the KVStore service APIs.
+//
+// The service shall be registered via app.KVStoreKey service key.
+type KVStoreService interface {
+	SetPluginKeyWithOptions(pluginID string, key string, value []byte, options model.PluginKVSetOptions) (bool, *model.AppError)
```

Comment:
This was actually a part of `ClusterService` interface in `L:53`, If you think it's best to separate these, would you mind removing it from that interface? Or, alternatively, you can use that service instead.